### PR TITLE
 Enhancements to "checking for invalidation"

### PR DESCRIPTION
**Pull request by RoyalSix**
#### Describe what your pull request addresses (Please include relevant issue numbers):
- This PR addresses bug with invalidation checking

#### Please include detailed Test instructions for your pull request:
- Copy steps in https://github.com/unfoldingword-dev/translationcore/issues/5941#issuecomment-490533836 and ensure they are all fixed

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6122)

### CODE REVIEW

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (6323764)](https://tc-builds.door43.org/build/feature-jay-5941/translationCore-linux-x64-1.2.0-6323764.deb) is ready for download.
[macOS build 1.2.0 (6323764)](https://tc-builds.door43.org/build/feature-jay-5941/translationCore-macos-x64-1.2.0-6323764.dmg) is ready for download.
[Windows x64 build 1.2.0 (6323764)](https://tc-builds.door43.org/build/feature-jay-5941/translationCore-win-x64-1.2.0-6323764.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (6323764)](https://tc-builds.door43.org/build/feature-jay-5941/translationCore-win-x32-1.2.0-6323764.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (17bec16)](https://tc-builds.door43.org/build/feature-jay-5941/translationCore-linux-x64-1.2.0-17bec16.deb) is ready for download.
[macOS build 1.2.0 (17bec16)](https://tc-builds.door43.org/build/feature-jay-5941/translationCore-macos-x64-1.2.0-17bec16.dmg) is ready for download.
[Windows x64 build 1.2.0 (17bec16)](https://tc-builds.door43.org/build/feature-jay-5941/translationCore-win-x64-1.2.0-17bec16.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (17bec16)](https://tc-builds.door43.org/build/feature-jay-5941/translationCore-win-x32-1.2.0-17bec16.setup.exe) is ready for download.

File: src/js/actions/ToolActions.js [hunk 1]
```diff
@@ -44,47 +44,49 @@ export const loadTools = (toolsDir) => (dispatch) => {
...
+        const language = getToolGatewayLanguage(getData(), name);
+        const groupIndex = loadProjectGroupIndex(language, name, projectDir, translate);
+        dispatch(loadGroupsIndex(groupIndex));
 
-      // verify stuff. We need this to pick up external edits and when checks data is updated.
-      // TRICKY: this must be after loadCurrentContextId() for group data changes to be saved to file
-      dispatch(GroupsDataActions.verifyGroupDataMatchesWithFs());
+        dispatch(GroupsDataActions.verifyGroupDataMatchesWithFs());
+        dispatch(loadCurrentContextId());
+        //TRICKY: need to verify groups data before and after the contextId has been loaded
```
Comment (mannycolon) [inline src/js/actions/ToolActions.js:80]: Why not only once and after loading the contextId?

**Comment by mannycolon** [comment]
Local Test: `LGTM`

File: src/js/actions/ToolActions.js (hunk 1 above)
Comment (mannycolon) [reply src/js/actions/ToolActions.js:80]: Why can't we verify groups data once just after loading the context id?

**Review by mannycolon — CHANGES_REQUESTED** [review]

**Review by RoyalSix — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6122)* status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @mannycolon)

___
*[src/js/actions/ToolActions.js, line 80 at r2](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6122#-LgYeW3TItbAhIMNjC0y-r2-80:-Lh1cJnI48fMEbqEejCJ:b-4p8yx8) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/17bec16d728c42281ec46540683becb441662428/src/js/actions/ToolActions.js#L80)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

Why can't we verify groups data once just after loading the context id?
</blockquote></details>

Some reason if I only call that method after it will not work. There is definitely something more complex about why this isn't working like it should but I was unable to figure it out. If I remember correctly the underlying issue was that I could not determine once a change was written to the FS from the redux subscriber method so some other action that reads from the FS was using old data. I also tried using a timeout but that did not work. This was the best I could come up with for now. This will only get called twice on the tool loading.

**Comment by mannycolon** [comment]
@RoyalSix I guess we'll have to address that somewhere else or at another time

**Review by mannycolon — APPROVED** [review]

**Pull request merged**